File: app/src/main/java/codes/wokstym/cookingrecipes/components/IngredientAdapter.java
package codes.wokstym.cookingrecipes.components;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;

import codes.wokstym.cookingrecipes.R;
import codes.wokstym.cookingrecipes.models.IngredientDto;

public class IngredientAdapter extends ArrayAdapter<IngredientDto> {
    public IngredientAdapter(Context context, List<IngredientDto> objects) {
        super(context, 0, objects);
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        IngredientDto ingredient = getItem(position);
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.item_ingredient, parent, false);
        }

        TextView name = convertView.findViewById(R.id.ingredient_name);
        TextView amount = convertView.findViewById(R.id.ingredient_amount);
        TextView unit = convertView.findViewById(R.id.ingredient_unit);


        name.setText(ingredient.getName());
        amount.setText(Integer.toString(ingredient.getAmount()));
        unit.setText(ingredient.getUnit().getName());

        return convertView;
    }

}

File: app/src/main/java/codes/wokstym/cookingrecipes/components/IngredientAdapter.kt
package codes.wokstym.cookingrecipes.components

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import codes.wokstym.cookingrecipes.databinding.ItemIngredientBinding
import codes.wokstym.cookingrecipes.models.IngredientDto

class IngredientAdapter(
        private val context: Context,
        private val objects: List<IngredientDto>) : RecyclerView.Adapter<IngredientAdapter.IngredientViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): IngredientViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val binding: ItemIngredientBinding = ItemIngredientBinding.inflate(inflater, parent, false)
        return IngredientViewHolder(binding)
    }

    override fun getItemCount(): Int = objects.size

    override fun onBindViewHolder(holder: IngredientViewHolder, position: Int) = holder.bind(objects[position])

    inner class IngredientViewHolder(private val binding: ItemIngredientBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(ingredient: IngredientDto) {
            binding.ingredientAmount.text = ingredient.amount.toString()
            binding.ingredientName.text = ingredient.name
            binding.ingredientUnit.text = context.getString(ingredient.unit.nameResourceId)
        }
    }
}